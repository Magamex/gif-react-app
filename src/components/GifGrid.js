import React from 'react'
// import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {

    // const [images, setImages] = useState([]);

    // useEffect(() => {
    //     getGifs(category)
    //         .then(setImages)
    // }, [category]);

    return (
        // <div className='card-grid'>
        //     <h3>{category}</h3>
        //         {
        //             images.map(img =>(
        //                <GifGridItem key={img.id} {...img}/> 
        //             ))
        //         }
        // </div>
    )
}
