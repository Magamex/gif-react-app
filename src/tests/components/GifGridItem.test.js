import React from 'react';

import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem/>', () => {

    const title = 'titulo';
    const url = 'https://media1.giphy.com/media/l2JhxQy0xZxZyQ7XS/200w.gif';
    const wrapper = shallow(<GifGridItem title={title} url={url}/>);

    test('Debe mostrar el componente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrar el title', () => {
        expect(wrapper.find('p').text().trim()).toBe(title)
    });

    test('Debe obtener la imagen y el alt', () => {
        const propiedades = wrapper.find('img').props();

        expect(propiedades.src).toBe(url);
        expect(propiedades.alt).toBe(title);
    });

    test('Debe tener animated__fadeIn', () => {
        const classname = wrapper.find('div').prop('className');
        expect(classname.includes('animate__fadeIn')).toBe(true);

    });
});
