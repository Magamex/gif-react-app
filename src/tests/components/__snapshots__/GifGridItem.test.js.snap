// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem/> Debe mostrar el componente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="titulo"
    src="https://media1.giphy.com/media/l2JhxQy0xZxZyQ7XS/200w.gif"
  />
  <p>
    titulo
  </p>
</div>
`;
